cmd: darwin build fails with undefined maybeRunAsService/sigHandlerConfigReload

:lady_beetle: **Describe the bug**

Building the server-enabled CLI on macOS fails due to missing Darwin-specific symbol implementations in the `cmd` package.

`cmd/serve.go` unconditionally calls `maybeRunAsService` and `sigHandlerConfigReload`, but neither is available for Darwin builds, causing compilation to fail.

:computer: **Components impacted**

ntfy server (CLI build path), specifically `make cli-darwin-server`

:bulb: **Screenshots and/or logs**

Reproduction command:

```bash
make cli-darwin-server
```

Build output:

```
# heckel.io/ntfy/v2/cmd
cmd/serve.go:509:26: undefined: maybeRunAsService
cmd/serve.go:517:5: undefined: sigHandlerConfigReload
make: *** [cli-darwin-server] Error 1
```

:crystal_ball: **Additional context**

Root cause appears to be build tags:

`cmd/serve.go` calls both functions unconditionally.
`cmd/serve_windows.go` only builds on `windows && !noserver`.
`cmd/serve_unix.go` only builds on `linux || dragonfly || freebsd || netbsd || openbsd` (Darwin missing).

This likely regressed when Windows service support was introduced (commit 6d5cc6ae).

--- a/cmd/serve_unix.go
+++ b/cmd/serve_unix.go
@@ -1,4 +1,4 @@
-//go:build linux || dragonfly || freebsd || netbsd || openbsd
+//go:build darwin || linux || dragonfly || freebsd || netbsd || openbsd
 
 package cmd
 

